### PR TITLE
Maid 862 get key messages

### PR DESCRIPTION
**Pull request by benjaminbollen**
Ported from MAID-844_Sentinel.  Needs continued work andd merging in work from Mahmoud

### CODE REVIEW

File: src/maidsafe/routing/sentinel.cc.autosave [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  if (node_key_accumulator_.HaveName(header.FromNode())) {  // ok direct
+  } else if (header.FromGroup() && group_key_accumulator_.HaveName(*header.FromGroup())) {  // ok dht
+  }
+  return boost::none;
+}
+
+}  // namespace routing
+
+}  // namespace maidsafe
```
Comment (chandraprakash) [inline src/maidsafe/routing/sentinel.cc.autosave]: delete this file plz

**Comment by chandraprakash** [comment]
this one is .autosave so will not overlap with anyone's work :-) You accedently added it :-)

**Comment by benjaminbollen** [comment]
I know, I already had it removed once, and it was there again.  Bad habit of git add --all developing :).  No I mean, this PR is far from what it should be.  The messages are largely boilerplate copied and then some minimal changes to accommodate for the message type, but it's not final at all yet.

Additionally, it also contains changes to sentinel.h etc.  So after Sentinel tests I will make a clean PR for this, but for now just serves for collaboration purposes.

**Comment by chandraprakash** [comment]
:+1:

**Pull request closed without merging**